src/main/java/calculator/domain/CalculateNumbers.java

```diff
@@ -0,0 +1,45 @@
+package calculator.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CalculateNumbers {
+    private static final String VALID_EMPTY_NUMBERS_MESSAGE = "입력 값이 없습니다.";
+    private static final String VALID_MIN_SIZE_MESSAGE = "공백을 기준으로 최소 3개 이상 입력바랍니다.";
+    private static final int MIN_SIZE = 3;
+    private final List<CalculateNumber> numbers;
+
+    public CalculateNumbers(String[] numbers) {
+        validNumber(numbers);
+        this.numbers = mapCaculateNumber(numbers);
+    }
+
+    private List<CalculateNumber> mapCaculateNumber(String[] numbers) {
+        List<CalculateNumber> calculateNumbers = new ArrayList<>();
+        for (int i = 0; i < numbers.length; i += 2) {
+            calculateNumbers.add(new CalculateNumber(numbers[i]));
+        }
+        return calculateNumbers;
+    }
+
+    private void validNumber(String[] numbers) {
+        validEmpty(numbers);
+        validMinSize(numbers);
+    }
+
+    private void validEmpty(String[] numbers) {
+        if (numbers == null || numbers.length == 0) {
+            throw new IllegalArgumentException(VALID_EMPTY_NUMBERS_MESSAGE);
+        }
+    }
+
+    private void validMinSize(String[] numbers) {
+        if (numbers.length > 0 && numbers.length < MIN_SIZE) {
+            throw new IllegalArgumentException(VALID_MIN_SIZE_MESSAGE);
+        }
+    }
+
+    public List<CalculateNumber> numbers() {
+        return this.numbers;
```

Comment:
final 키워드를 붙이더라도, 그 안의 리스트 요소를 추가하는 부분이 방지되지는 않을것 같아요.

아래와 같이 반환하면 어떨까요?

```suggestion
        return Collections.unmodifiableList(this.numbers);
```